Meetme.MatchingService/Meetme.MatchingService.Application/Services/Match/MatchService.cs

```diff
@@ -10,19 +12,23 @@ public class MatchService : IMatchService
 	private readonly IProfileServiceClient _profileServiceClient;
 	private readonly IRepository<MatchEntity> _matchRepository;
 	private readonly IRepository<LikeEntity> _likeRepository;
+	private readonly IPublisher _mediator;
+	private readonly IMongoRepository _mongoRepository;
 
-	public MatchService(IProfileServiceClient profileServiceClient, IRepository<MatchEntity> matchRepository, IRepository<LikeEntity> likeRepository)
+	public MatchService(IProfileServiceClient profileServiceClient, IRepository<MatchEntity> matchRepository, IRepository<LikeEntity> likeRepository, IMediator mediator, IMongoRepository mongoRepository)
 	{
 		_profileServiceClient = profileServiceClient;
 		_matchRepository = matchRepository;
 		_likeRepository = likeRepository;
+		_mediator = mediator;
+		_mongoRepository = mongoRepository;
 	}
 
 	public async Task MatchProfilesAsync(LikeModel likeModel, CancellationToken cancellationToken)
 	{
-		var areProfilesMatched = await TryMatchProfilesAsync(likeModel, cancellationToken);
+		var (areProfilesMatched, profile, matchedProfile) = await TryMatchProfilesAsync(likeModel, cancellationToken);
```

Comment:
if matched profile != null this means that areProfilesMatched == true (and vice versa). So, this flag seems to be redundant